docs/GOVERNANCE.md

```diff
@@ -0,0 +1,207 @@
+- **Maintained by**: Component owners
+- **Decisions by**: Component owners
+- **Support**: Best-effort community support
+
+## Tier Transitions
+
+*Process for moving components between Core and Third-Party tiers.*
+
+### Promotion to Core
+1. **Nomination**: Any maintainer can nominate
+2. **Review**: Technical and governance review
+3. **Requirements**: Must meet all core tier requirements
+4. **Decision**: Steering committee approval
+5. **Timeline**: 4-6 weeks review process
+
+### Demotion from Core
+Triggers:
+- Maintenance neglect (>6 months)
+- Security issues unaddressed
+- Breaking changes without migration
+- Community consensus
+
+## Removal Policies
+
+*Timeline and criteria for removing inactive or problematic components from the repository.*
+
+### Verification Process (9 months)
+Components are marked for verification if:
+- No updates in 9 months
+- Maintainers unresponsive
+- Compatibility issues
+
+### Removal Process (12 months)
+After 12 months of inactivity:
+1. **Notice**: 30-day removal notice
+2. **Community input**: 2-week feedback period
+3. **Final decision**: Steering committee
+4. **Removal**: Delete component code from repository
+
+### Emergency Removal
+Immediate removal for:
+- Critical security vulnerabilities
+- Legal issues
+- Malicious code
+
+## Deprecation Policy
+
+*Structured approach to deprecating components with adequate notice and migration support.*
+
+### Two-Release Policy
+Components deprecated for 2 Kubeflow releases before removal.
+
+**Process:**
+1. **Deprecation notice**: Mark as deprecated
+2. **Migration guide**: Provide alternatives
+3. **Community notice**: Announce in releases
+4. **Removal**: After 2 releases
+
+## Repository Roles
+
+*Key roles and responsibilities for governing and maintaining the repository.*
+
+### Repository Maintainers
+- **Role**: Day-to-day repository management
+- **Responsibilities**: Reviews, releases, community support
+- **Requirements**: Active contributor, community trust
+
+### Component Owners
+- **Role**: Individual component maintenance
+- **Responsibilities**: Updates, bug fixes, user support
+- **Requirements**: Technical expertise, time commitment
+
+### Kubeflow Steering Committee Member
+- **Role**: Organization-scope strategic decisions and governance
+- **Responsibilities**: Defined in the organization-wide [Kubeflow governance documentation](https://www.kubeflow.org/docs/about/governance/#2-kubeflow-steering-committee-ksc)
+
+## Decision Making
+
+*Framework for making technical, policy, and strategic decisions within the community.*
+
+### Decision Types
+- **Technical**: Component owners → maintainers → steering committee
+- **Policy**: Maintainers → steering committee
+- **Strategic**: Steering committee
+
+### Process
+1. **Proposal**: Create GitHub issue/RFC
+2. **Discussion**: Community feedback (1-2 weeks)
+3. **Decision**: Appropriate authority level
+4. **Implementation**: Assign and track
+
+## Conflict Resolution
```

Comment:
this is bloating the doc, I feel like this section is not needed, let's try to keep the doc short and concise - it's already too long IMO